### PR TITLE
[tls] Add a PEMTrustManager to deal with different PEM files (e.g. self-signed or global CA certificates)

### PR DESCRIPTION
**Pull request by cweitkamp**
- Added a PEMTrustManager to deal with different PEM files (e.g. self-signed or global CA certificates)

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+    /**
+     * Creates a {@link PEMTrustManager} instance by passing the PEM certificate as {@link String}.
+     * The PEM format typically starts with <code>"-----BEGIN CERTIFICATE-----"</code> and ends with
+     * <code>"-----END CERTIFICATE-----"</code>. The base 64 encoded certificate information are placed in between.
+     *
+     * @param pemCert the PEM certificate
+     * @throws CertificateException
+     */
+    public PEMTrustManager(String pemCert) throws CertificateException {
+        if (!pemCert.isBlank() && pemCert.startsWith(BEGIN_CERT)) {
```
Comment (J-N-K) [inline bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java:69]: One of the use-cases (e.g. with the http binding) could be that there is a configuration field in the thing config that allows pinning the certificate permanently (the `getInstanceFromServer` does only pin until restarting the system). 

Using `line.separator` would require the configuration  to be different depending on the operating system that openHAB runs on and could lead to problems if the user configures on a different OS. Maybe the cert string should be normalized to the `line.separator` line endings before the checks are done.

File: bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+    public static PEMTrustManager getInstanceFromFile(String path) throws FileNotFoundException, CertificateException {
+        String pemCert = readPEMCertificateStringFromFile(path);
+        if (pemCert != null) {
+            return new PEMTrustManager(pemCert);
+        }
+        throw new CertificateInstantiationException(String.format("Unable to read certificate from file: %s", path));
+    }
+
+    /**
+     * Creates a {@link PEMTrustManager} instance by downloading the PEM certificate from the given file.
```
Comment (J-N-K) [inline bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java:98]:
```suggestion
     * Creates a {@link PEMTrustManager} instance by downloading the PEM certificate from the given server.
```

File: bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+        }
+        return null;
+    }
+
+    private static @Nullable String readPEMCertificateStringFromFile(String path) throws FileNotFoundException {
+        File certFile = new File(path);
+        if (certFile.exists()) {
+            try {
+                String cert = new String(Files.readAllBytes(certFile.toPath()), StandardCharsets.UTF_8);
+                if (cert.startsWith(BEGIN_CERT)) {
```
Comment (J-N-K) [inline bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java:195]: You check the same at the beginning of the constructor, isn't that enough?

**Comment by J-N-K** [comment]
This is IMO one of the most useful extensions in the last years, it allows the remove the `TrustAllTrustManager` and similar implementations in quite a bunch of addons.

**Review by cweitkamp — COMMENTED** [review]
Good to see that someone finds it very useful.

File: bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java (hunk 1 above)
Comment (cweitkamp) [reply bundles/org.openhab.core.io.net/src/main/java/org/openhab/core/io/net/http/PEMTrustManager.java:68]: Can you check if the new version handles the line endings even better?

**Review by kaikreuzer — APPROVED** [review]
lgtm, thank you!
Let's wait for @J-N-K final comments before merging.

**Comment by J-N-K** [comment]
I'll check today and report back this evening.

**Review by J-N-K — APPROVED** [review]
LGTM

**Pull request merged**